### PR TITLE
Make scheduler interfaces private

### PR DESCRIPTION
**Pull request by ktbarrett**
Most of the scheduler methods which are currently "public" (don't start with an underscore) are not safe or reasonable for a user to call under any circumstance and were definitely intended to be private. Stubs remain in place of the original name to throw deprecation warnings. The interfaces not privatized are:

* `add`, same thing as `cocotb.fork`.
* `create_task` , is a staticmethod so it is safe to call whenever.
* `start_soon`, designed to be user facing.

It is intended that the stubs will be removed in 2.0.

xref #2278.

### CODE REVIEW

File: cocotb/scheduler.py [hunk 1]
```diff
@@ -390,13 +399,13 @@ def _event_loop(self, trigger):
                 trigger = self._pending_triggers.pop(0)
 
                 if not is_first and isinstance(trigger, GPITrigger):
-                    self.log.warning(
+                    self.log.warnings(
```
Comment (garmin-mjames) [inline cocotb/scheduler.py:402]: Inadvertent?
```suggestion
                    self.log.warning(
```

File: cocotb/scheduler.py [hunk 2]
```diff
@@ -390,13 +399,13 @@ def _event_loop(self, trigger):
...
                         "A GPI trigger occurred after entering react - this "
                         "should not happen."
                     )
                     assert False
 
-                # this only exists to enable the warning above
+                # this only exists to enable the warnings above
```
Comment (garmin-mjames) [inline cocotb/scheduler.py:408]: ?
```suggestion
                # this only exists to enable the warning above
```

File: cocotb/scheduler.py [hunk 3]
```diff
@@ -326,6 +327,14 @@ def _test_completed(self, trigger=None):
             self._check_termination()
 
     def react(self, trigger):
+        """
+        .. deprecated:: 1.5
+            This function is now private
```
Comment (garmin-mjames) [inline cocotb/scheduler.py:332]: Similarly for others, to be more consistent with the rest of the docs.
```suggestion
        .. deprecated:: 1.5
            This function is now private.
```

Comment (garmin-mjames) [reply cocotb/scheduler.py:332]: Was this missed or do you prefer no period?

Comment (ktbarrett) [reply cocotb/scheduler.py:332]: Just forgot to save before pushing.

**Comment by ktbarrett** [comment]
Should I create a newsfragment for #2278? I don't think I should list every change because in the end there may be a lot.

**Comment by garmin-mjames** [comment]
I think a summary newsfragment would be good. We can link to the individual PRs from that issue if we want to track all of the changes.

**Review by garmin-mjames — APPROVED** [review]

File: documentation/source/newsfragments/2278.removal.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+The access mode of many interfaces in the cocotb core libraries were re-evaluated.
```
Comment (cmarqu) [inline documentation/source/newsfragments/2278.removal.rst:1]:
```suggestion
The access modes of many interfaces in the cocotb core libraries were re-evaluated.
```
?

**Pull request merged**